modules/data/report.py

```diff
@@ -0,0 +1,188 @@
+                                        table_name="user_report",
+                                        where="user_id = %s",
+                                        where_args=(user_id, ),
+                                        order_by="message_date DESC")
+
+        if len(reports) == 0:  # the vote is not present
+            return None
+        report = reports[0]
+        return cls(user_id=report['user_id'],
+                   target_username=report['target_username'],
+                   date=datetime.strptime(report['message_date'], "%Y-%m-%d %H:%M:%S.%f"),
+                   group_id=report['group_id'],
+                   g_message_id=report['g_message_id'])
+
+    @classmethod
+    def from_group(cls, group_id: int, g_message_id: int):
+        """Gets a report of any type related to the specified message in the admin group
+
+        Args:
+            group_id (int): id of the admin group
+            g_message_id (int): id of the report in the group
+
+        Returns:
+            Report: istance of the class or None if the report was not present
+        """
+        reports = DbManager.select_from(select="*",
+                                        table_name="user_report",
+                                        where="group_id = %s and g_message_id = %s",
+                                        where_args=(group_id, g_message_id))
+
+        if len(reports) > 0:  # the vote is not present
+            report = reports[0]
+            return cls(user_id=report['user_id'],
+                       target_username=report['target_username'],
+                       date=datetime.strptime(report['message_date'], "%Y-%m-%d %H:%M:%S.%f"),
+                       group_id=report['group_id'],
+                       g_message_id=report['g_message_id'])
+
+        reports = DbManager.select_from(select="*",
+                                        table_name="spot_report",
+                                        where="group_id = %s and g_message_id = %s",
+                                        where_args=(group_id, g_message_id))
+
+        if len(reports) > 0:  # the vote is not present
+            report = reports[0]
+            return cls(user_id=report['user_id'],
+                       c_message_id=report['c_message_id'],
+                       group_id=report['group_id'],
+                       g_message_id=report['g_message_id'])
+
+        return None
+
+    def __repr__(self):
+        if self.c_message_id is not None:
+            return f"PostReport: [ user_id: {self.user_id}\n"\
+                    f"c_message_id: {self.c_message_id}\n"\
+                    f"group_id: {self.group_id}\n"\
+                    f"g_message_id: {self.g_message_id} ]"
+        else:
+            return f"UserReport: [ user_id: {self.user_id}\n"\
+                    f"target_username: {self.target_username}\n"\
+                    f"group_id: {self.group_id}\n"\
+                    f"g_message_id: {self.g_message_id} ]"
```

Comment:
"Maybe" you could use a variable here
```suggestion
    def __repr__(self):
        target = f"c_message_id: {self.c_message_id}\n" if self.c_message_id is not None else f"group_id: {self.group_id}\n"
          
        return f"UserReport: [ user_id: {self.user_id}\n"\
                target \
                f"group_id: {self.group_id}\n"\
                f"g_message_id: {self.g_message_id} ]"
```

or just put the ternary condition inside the return.

What do you think? :thinking:

Reply:
Welp, first of all i forgot to add the date field to the user report.
Anyhow, even if that way it would be more compact, i kinda like the extreme readability of this approach, and i think it's appropriate for a method you are supposed to use when you want to better understand the class or something is not behaving as expected

Reply:
ok ;-)